src/matrices/skeletonization.jl

```diff
@@ -0,0 +1,396 @@
+#=
+In this file the core computations for skeletonizing a recurrence matrix
+are defined.
+=#
+
+# Core function, which gets exported
+"""
+    skeletonize(R) → R_skel
+
+Skeletonizes the recurrence matrix `R` (see [`recurrence_matrix`](@ref)) by using
+the algorithm proposed by Kraemer & Marwan [^Kraemer2019]. This function returns
+`R_skel`, a recurrence matrix, which only consists of diagonal lines of
+"thickness" one.
+
+## References
+[^Kraemer2019]: Kraemer, K.H., Marwan, N. (2019). [Border effect corrections for diagonal line based recurrence quantification analysis measures. Physics Letters A 383(34)](https://doi.org/10.1016/j.physleta.2019.125977).
+"""
+function skeletonize(X::Union{ARM,SparseMatrixCSC})
+
+    if issymmetric(X)
+        symm = true
+        # convert lower triangle into a close returns map
+        X_cl1 = convert_recurrence_matrix(tril(X))
+        # get "horizontal" line distribution with position indices
+        lines1t, lines1l, lines1c = horizontalhisto(X_cl1)
+        lines_copy1t = deepcopy(lines1t)
+        lines_copy1l = deepcopy(lines1l)
+        lines_copy1c = deepcopy(lines1c)
+
+        # create a close returns map with horizontal lines represented by
+        # numbers, equal to their lengths
+        X_hori1 = create_close_returns_map(lines_copy1t, lines_copy1l, lines_copy1c, size(X_cl1))
+    else
+        symm = false
+        # convert upper and lower triangles into close returns maps
+        X_cl1 = convert_recurrence_matrix(tril(X))
+        X_cl2 = convert_recurrence_matrix(sparse(triu(X)'))
+
+        # get "horizontal" line distribution with position indices
+        lines1t, lines1l, lines1c = horizontalhisto(X_cl1)
+        lines_copy1t = deepcopy(lines1t)
+        lines_copy1l = deepcopy(lines1l)
+        lines_copy1c = deepcopy(lines1c)
+        lines2t, lines2l, lines2c = horizontalhisto(X_cl2)
+        lines_copy2t = deepcopy(lines2t)
+        lines_copy2l = deepcopy(lines2l)
+        lines_copy2c = deepcopy(lines2c)
+
+        # create close returns maps with horizontal lines represented by
+        # numbers, equal to their lengths
+        X_hori1 = create_close_returns_map(lines_copy1t, lines_copy1l, lines_copy1c, size(X_cl1))
+        X_hori2 = create_close_returns_map(lines_copy2t, lines_copy2l, lines_copy2c, size(X_cl2))
+    end
+
+    # scan the lines, start with the longest one and discard all adjacent lines
+    lines_final_t, lines_final_l, lines_final_c = get_final_line_matrix(lines1t, lines1l, lines1c, lines_copy1t, lines_copy1l, lines_copy1c, X_hori1)
+
+    # if not symmetric input RP, than compute for the upper triangle as well
+    if ~symm
```

Comment:
```suggestion
    if !symm
```

The `!` is the typical operator used for "not".

Reply:
done